server/controllers/bookController.js

```diff
@@ -134,6 +136,63 @@ class BookController {
       Utils.errorStatus(res, 500, error.message);
     }
   }
+
+  static async recieveBook(req, res){
+    const { isbn, patronId } = req.body;
+    const { damaged } = req.params;
+    let fineCost = 0;
+    let returnMessage = null;
+    try {
+      const borrowedBook = await models.BorrowedBooks.findOne({
+        where: {
+          [Op.and]: [{ isbn }, { patronId }]
+        }
+      });
+      if(!borrowedBook){
+        return Utils.errorStatus(res, 404, 'Book was not found');
+      }
+      if(borrowedBook.dueDate < new Date()) {
+        fineCost = 200;
```

Comment:
Well done @cvjude I see you implemented a lot of checks in this controller. One suggestion on though, maybe in your next task you can modify this so that the fine increases based on the number of days past the due date.

Reply:
Nice idea, I'll get on it asap